### PR TITLE
Support Vault Auth method

### PR DESCRIPTION
**Pull request by Kryvchun**
This PR adds support for vault Kubernetes authentication.

Issue: https://github.com/hashicorp/envconsul/issues/274.

New parameters:
```
  -vault-k8s-auth-role-name
      Enable k8s auth method and use this role name

  -vault-k8s-service-account-token
      Use this service token for the k8s auth method

  -vault-k8s-service-account-token-path
      Use the service token from the file for the k8s auth method

  -vault-k8s-service-mount-path
      Use this login mount path for the k8s auth method
```

### CODE REVIEW

**Review by ygworldr — APPROVED** [review]
LGTM

**Comment by Kryvchun** [comment]
@eikenb Hi, it is ready for review.

**Comment by eikenb** [comment]
With the release of consul-template 0.29.1 that included the changes needed for this and the dependency has already been updated.

IE. I'll be reviewing this soon for inclusion in 0.13.0.

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,8 @@
 /pkg
 /vendor
 envconsul
+config.hcl
```
Comment (eikenb) [inline .gitignore:6]: I don't mind adding these to the .gitignore but I prefer keeping unrelated changes out of feature PRs. Would you mind removing this? Please feel free to submit it as a standalone PR if you'd like (I'll just merge it).

File: README.md [hunk 2]
```diff
@@ -451,6 +451,28 @@ vault {
...
+  # to authenticate Vault using a Kubernetes Service Account Token.
+  # This method of authentication makes it easy to introduce a Vault
+  # token into a Kubernetes Pod.
+  # The default value is "".
+  k8s_auth_role_name = ""
+
+  # This is the path of file that contains service token for k8s auth method.
+  # It will be ignored if "k8s_service_account_token" set.
+  # The default value is "/run/secrets/kubernetes.io/serviceaccount/token".
+  k8s_service_account_token_path = "/run/secrets/kubernetes.io/serviceaccount/token"
```
Comment (eikenb) [inline README.md:465]: Nice! I like having the default value as the documented example value.

**Review by eikenb — APPROVED** [review]
With the one change (.gitignore) this looks good.

Thanks!

File: .gitignore (hunk 1 above)
Comment (Kryvchun) [reply .gitignore]: Fixed!

**Comment by Kryvchun** [comment]
@eikenb 
`.gitignore` fixed and `main` merged.
Thank you

**Pull request merged**